merlin/systems/dag/ops/fil.py

```diff
@@ -512,13 +517,12 @@ def num_targets(self):
 class CUMLRandomForest(FILModel):
 
     model_type = "treelite_checkpoint"
-    model_filename = "model.pkl"
+    model_filename = "checkpoint.tl"
 
     def save(self, version_path):
         """Save model to version_path."""
         model_path = pathlib.Path(version_path) / self.model_filename
-        with open(model_path, "wb") as model_file:
-            pickle.dump(self.model, model_file)
+        self.model.convert_to_treelite_model().to_treelite_checkpoint(str(model_path))
```

Comment:
we haven't currently got `cuml` installed in the test env, so this is untested in CI currently